run.sh

```diff
@@ -39,6 +39,20 @@ catch_failed () {
     exit 1
 }
 
+# detect the device
+device=""
+for d in /dev/cdrom /dev/sr[0-9]*; do
+    if [ -b "$d" ]; then
+        device="$d"
+        break
+    fi
+done
+if [ -z "$device" ]; then
+    echo -e "\nERROR: Unable to find optical device to eject.\n" >&2
+    exit 1
+fi
+echo "device=$device"
```

Comment:
set +x should take care of printing the values of variables so no need for these echos.